### ISSUE TITLE
Inverse the sorting of permissions an roles in sandstorm-pkgdef.capnp

### ISSUE THREAD

**Issue by alexander-vidoni** [sandstorm-io/sandstorm#2613]
In *sandstorm-pkgdef.capnp* the sample permission array is sorted from *high* permissions to *low* permissions and the roles follow in the same order.

When you open the ShareAccess dialog in the top menue you find the roles in the same order as they are specified in the config file (sandstorm-pkgdef.capnp).

The problem:

If you want to share a grain and you are in a hurry it is too easy to create a link or an email link with the most prvileged role. 

Solution:

List the permission and roles (in sandstorm-pkgdef.capnp) in *ascending* order. 
If you do so, you only could grant the lowest privilege by mistake than the highest.

Thanks for reading.

**Comment by ocdtrekkie** [comment]
Hmm... I think this could go either way. Like for a shared Etherpad or Wekan document, I feel most users would intend to share edit rights, which is the highest shareable permission. And I think those document-type apps with collaborative editing are probably the biggest use of Sandstorm I see personally. But I suppose for Davros, one is more likely to read-only share their files, and if applications like MediaWiki expose an admin role, it'd be bad to default to that.

**Comment by kentonv** [comment]
Hi Alexander,

Apps can actually specify which role should be the default in the sharing dialog by setting `default = true` on one of the roles. But I agree with @ocdtrekkie that it's not entirely clear what the default role should be. With Etherpad I intentionally chose "edit" as the default because that's almost always what I want in practice.

Note that you cannot reorder the roles after an app has been published -- doing so will break compatibility with existing shares, since the role granted by a share is stored by numeric index.

**Comment by alexander-vidoni** [comment]
I agree and I know that the design of the roles and permissons is in the hands of each app developer. Reordering the apps permissions and roles has unwanted side effects for existing apps -> OK ( I won't do it for existing app-instances :-) ).

I just wanted to show that the example/template in ***sandstorm-pkgdef.capnp*** has the potential to guide a new developer (and I am new to sandstorm.io) to arrange the privileges in a manner that could lead to a potentially unsafer daily usage.

> The safest fortress/instance/grain is useless if the owner gives the master key away by mistake.

IMHO it would be the safer way to grant lower rights by mistake than higher. A later upgrade to higher permissions or roles is more harmless than trying to get leaked data back.

I focus on this aspect because I am designing my first app and one of my first decisions is to define the different roles. This is basic work and difficult to change later in the life cycle of the app. 

I think it would be sufficient to add this perspective to the comments in ***sandstorm-pkgdef.capnp***, so each developer could deside on its own.
